subprojects/plugins/src/main/java/org/gradle/api/plugins/JvmTestSuitePlugin.java

```diff
@@ -73,14 +74,24 @@ public void apply(Project project) {
         ExtensiblePolymorphicDomainObjectContainer<TestSuite> testSuites = testing.getSuites();
         testSuites.registerBinding(JvmTestSuite.class, DefaultJvmTestSuite.class);
 
-        // TODO: Deprecate this behavior?
-        // Why would any Test task created need to use the test source set's classes?
         project.getTasks().withType(Test.class).configureEach(test -> {
             // The test task may have already been created but the test sourceSet may not exist yet.
             // So defer looking up the java extension and sourceSet until the convention mapping is resolved.
             // See https://github.com/gradle/gradle/issues/18622
-            test.getConventionMapping().map("testClassesDirs", () -> JvmPluginsHelper.getDefaultTestSuite(project).getSources().getOutput().getClassesDirs());
-            test.getConventionMapping().map("classpath", () -> JvmPluginsHelper.getDefaultTestSuite(project).getSources().getRuntimeClasspath());
+            test.getConventionMapping().map("testClassesDirs", () -> {
+                DeprecationLogger.deprecate("Configuring the test classes dirs by default for standalone Test tasks")
```

Comment:
This is written in a confusing way.  The user isn't configuring anything.  Maybe...

> Relying on the convention for Test.testClassesDirs in custom Test tasks